node e2e node flake: test panicked

It failed with the following error:

I0517 07:58:40.157487     944 e2e_node_suite_test.go:94] ******************************************************
I0517 07:58:40.288216     944 e2e_node_suite_test.go:131] Locksmithd is masked successfully
I0517 07:58:43.304213     944 e2e_node_suite_test.go:72] Node services started.  Running tests...
panic: runtime error: integer divide by zero
[signal 0x8 code=0x1 addr=0x7ceeec pc=0x7ceeec]

goroutine 47 [running]:
panic(0x1b1ba60, 0xc82000e070)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
k8s.io/kubernetes/test/e2e/framework.(*containerResourceGatherer).getKubeSystemContainersResourceUsage(0xc82004ff20, 0xc8201e2140)
	/var/lib/jenkins/workspace/node-pull-build-e2e-test@2/go/src/k8s.io/kubernetes/test/e2e/framework/resource_usage_gatherer.go:208 +0x2c
k8s.io/kubernetes/test/e2e/framework.(*containerResourceGatherer).startGatheringData(0xc82004ff20)
	/var/lib/jenkins/workspace/node-pull-build-e2e-test@2/go/src/k8s.io/kubernetes/test/e2e/framework/resource_usage_gatherer.go:290 +0x29
created by k8s.io/kubernetes/test/e2e/framework.(*Framework).BeforeEach
	/var/lib/jenkins/workspace/node-pull-build-e2e-test@2/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:160 +0xbf


https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/25671/node-pull-build-e2e-test/6309/build-log.txt


@gmarek

--- a/test/e2e/framework/resource_usage_gatherer.go
+++ b/test/e2e/framework/resource_usage_gatherer.go
@@ -205,6 +205,9 @@ func getKubemarkMasterComponentsResourceUsage() ResourceUsagePerContainer {
 }
 
 func (g *containerResourceGatherer) getKubeSystemContainersResourceUsage(c *client.Client) {
+	if len(g.workers) == 0 {
+		return
+	}
 	delayPeriod := resourceDataGatheringPeriod / time.Duration(len(g.workers))
 	delay := time.Duration(0)
 	for i := range g.workers {

